src/main/java/org/spongepowered/api/item/inventory/types/GridInventory.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory.types;
+
+import com.flowpowered.math.vector.Vector2i;
+import com.google.common.base.Optional;
+
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.Slot;
+import org.spongepowered.api.item.inventory.transaction.InventoryOperationResult;
+
+/**
+ * An GridInventory is an {@link org.spongepowered.api.item.inventory.Inventory}
+ * which is ordered into a coherent grid format, meaning that its slots can be
+ * referred to by X-Y coordinates as well as single indices.
+ */
+public interface GridInventory extends Inventory2D {
+
+    /**
+     * Gets the number of columns in the inventory.
+     *
+     * @return The width of this ItemGrid.
```

Comment:
Looks like `ItemGrid` as renamed to `GridInventory` but the javadoc got left behind.
Doc is also outdated in `item.recipe.Recipe` too